libraries/AdaptiveExpressions/ExpressionFunctions.cs

```diff
@@ -1123,17 +1123,33 @@ private static (object value, string error) GetProperty(Expression expression, I
         {
             object value = null;
             string error;
-            object instance;
+            object firstItem;
             object property;
 
             var children = expression.Children;
-            (instance, error) = children[0].TryEvaluate(state, options);
+            (firstItem, error) = children[0].TryEvaluate(state, options);
             if (error == null)
             {
-                (property, error) = children[1].TryEvaluate(state, options);
-                if (error == null)
+                if (children.Length == 1)
                 {
-                    (value, error) = WrapGetValue(MemoryFactory.Create(instance), (string)property, options);
+                    // get root value from memory
+                    if (!(firstItem is string))
+                    {
+                        error = $"if GetProperty contains only one parameter, please make sure it is string. But get {firstItem?.GetType()?.Name ?? "null"}";
```

Comment:
>if GetProperty contains only one parameter, please make sure it is string. But get {firstItem?.GetType()?.Name ?? "null" [](start = 34, length = 120)

"Single parameter {firstItem} is not a string.

Reply:
# Solved